hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/SelectProcessorSupplier.java

```diff
@@ -151,4 +158,35 @@ public void readData(ObjectDataInput in) throws IOException {
         parameterPositions = in.readIntArray();
         dialectName = in.readString();
     }
+
+    private String isNumberTypeCheck(ResultSetMetaData metaData, String type, int column) throws SQLException {
```

Comment:
I don't really like the fact that Oracle-specific logic leaks to general SelectProcessorSupplier. Can you move it to HazelcastXyzDialect (by default will return type, in Oracle variant will do all the magic you added here)?

Reply:
Other solution: create a Map<String, FunctionEx> constant with mapping of dialectname -> function that adds checks like this above. Then you add mapping e.g. "Oracle" -> OracleSqlDialect::numberTypeCheck, where OracleSqlDialect::numberTypeCheck is a new method - isNumberTypeCheck moved to the dialect class. If no value is found in this new mapping field, then no additional actions are performed. WDYT @Edizc?